infrastructure/authors/authors_test.go

```diff
@@ -0,0 +1,102 @@
+package authors
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/exlibris-fed/exlibris/model"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetByID(t *testing.T) {
+	conn, mock, _ := sqlmock.New()
+	ts := time.Date(2020, 7, 11, 12, 0, 0, 0, time.UTC)
+	authorSourceRows := sqlmock.NewRows([]string{"created_at", "updated_at", "deleted_at", "open_library_id", "name"}).
+		AddRow(ts, ts, nil, "OL1234567A", "writer mcwriterface")
+	mock.ExpectQuery(regexp.QuoteMeta("SELECT * FROM \"authors\" WHERE \"authors\".\"deleted_at\" IS NULL AND ((open_library_id = $1)) ORDER BY \"authors\".\"open_library_id\" ASC LIMIT 1")).
+		WithArgs("OL1234567A").
+		WillReturnRows(authorSourceRows)
+	bookSourceRows := sqlmock.NewRows([]string{"created_at", "updated_at", "deleted_at", "open_library_id", "title", "published", "isbn", "description"}).
+		AddRow(ts, ts, nil, "OL1234567W", "title", 1563408000, 123456789, "description")
+	mock.ExpectQuery(regexp.QuoteMeta("SELECT * FROM \"books\" INNER JOIN \"book_authors\" ON \"book_authors\".\"book_open_library_id\" = \"books\".\"open_library_id\" WHERE \"books\".\"deleted_at\" IS NULL AND ((\"book_authors\".\"author_open_library_id\" IN ($1)))")).
+		WillReturnRows(bookSourceRows)
+
+	db, _ := gorm.Open("postgres", conn)
+
+	repo := New(db)
+
+	author, err := repo.GetByID("OL1234567A")
+	assert.NoError(t, err)
+	assert.NotNil(t, author)
+	assert.Equal(t, "writer mcwriterface", author.Name)
+	assert.NoError(t, mock.ExpectationsWereMet())
+}
+
+func TestGetByID_NotFound(t *testing.T) {
+	conn, mock, _ := sqlmock.New()
+	mock.ExpectQuery(regexp.QuoteMeta("SELECT * FROM \"authors\" WHERE \"authors\".\"deleted_at\" IS NULL AND ((open_library_id = $1)) ORDER BY \"authors\".\"open_library_id\" ASC LIMIT 1")).
```

Comment:
I'm worried about this being too tightly coupled, but also it's better than what we have now!